packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -478,7 +533,9 @@ LightningElement.prototype = {
             );
         }
 
-        return attachInternals(elm);
+        const internals = attachInternals(elm);
+        // #TODO[2970]: remove proxy once `UpgradeableConstructor` has been removed
+        return createElementInternalsProxy(internals, Boolean(formAssociated));
```

Comment:
Note `Boolean(formAssociated)` will return `true` for a `false` string but that's how it [works](https://jsfiddle.net/r3tbyezf/1/) with vanilla web components too.